packages/aws-rfdk/lib/deadline/lib/version.ts

```diff
@@ -72,6 +72,149 @@ abstract class VersionQueryBase extends Construct implements IVersion {
+  public static parse(version: string): Version {
+    if (!Version.validateVersionFormat(version)) {
+      throw new TypeError(`Invalid version format. Expected format 'a.b.c.d', found '${version}'`);
+    }
+
+    return new Version(version.split('.').map(x => parseInt(x)));
+  }
+
+  /**
+   * This method validates the given string for a sequence '.' separated numbers.
+   *
+   * @param version the string to be validated.
+   *
+   * @returns true if the format is correct, else false.
+   */
+  private static validateVersionFormat(version: string): boolean {
+    /**
+     * Regex: ^\d+(?:\.\d+){3}$
+     * Matches a sequence of '.' separated numbers with exactly 4 digits.
+     * - ^ asserts position at start of a line.
+     * - \d+ Matches one or more digits.
+     * - (?:\.\d+) Matches a dot and the following one or more digits.
+     * - {3} Matches previous pattern exactly 3 times.
+     * - $ asserts position at the end of a line
+     */
+    if (version.match(/^\d+(?:\.\d+){3}$/)) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Numeric components of version.
+   */
+  private readonly components: number[];
+
+  /**
+   * @inheritdoc
+   */
+  public get majorVersion(): number {
+    return this.components[0];
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public get minorVersion(): number {
+    return this.components[1];
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public get releaseVersion(): number {
+    return this.components[2];
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public get patchVersion(): number {
+    return this.components[3];
+  }
+
+  constructor(components: number[]) {
+    // validations
+    if(components.length != 4) {
+      throw new Error('Invalid version format. Version should contain exactly 4 components.');
+    }
+    components.forEach((component) => {
+      if (component < 0) {
+        throw new RangeError('Invalid version format. None of the version components can be negative.');
+      }
+    });
+
+    this.components = components;
+  }
+
+  /**
+   * This method compares two version strings
+   *
+   * @param version
+   *
+   * @returns true if this version is greater than the provided version;
+   * false if this version is less than or equal to the provided verison.
+   */
+  public isGreaterThan(version: Version): boolean {
+    if (this.components.length != version.components.length) {
+      throw new TypeError('Component count in both the versions should be same.');
+    }
+
+    for (let i = 0; i < this.components.length; i++) {
+      if (this.components[i] != version.components[i]) {
+        return this.components[i] > version.components[i];
+      }
+    }
+    return false;
+  }
+
+  /**
+   * This method compares two version strings
+   *
+   * @param version
+   *
+   * @returns true if this version is less than the provided version;
+   * false if this version is greater than or equal to the provided verison.
+   */
+  public isLessThan(version: Version): boolean {
```

Comment:
There is some code duplication between `isLessThan` and `isGreaterthan`.
I think in some earlier iteration, you already used an integer as return type in `Utils.versionCompare`.
Maybe we can use a something like below to remove the code duplication and to have all 3 comparisons available:

```
private static compare(first: Version, second: Version): number {
  if (first.components.length != second.components.length) {
    throw new TypeError('Component count in both the versions should be same.');
  }

  for (let i = 0; i < first.components.length; i++) {
    const diff = first.components[i] - second.components[i];
    if (diff != 0) {
      return diff;
    }
  }
  return 0;
}


public isGreaterThan(version: Version): boolean {
  return compare(this, version) == 1;
}

public isGreaterThan(version: Version): boolean {
  return compare(this, version) == -1;
}

public isEqual(version: Version): boolean {
  return compare(this, version) == 0;
}
```

Reply:
updated